amplipi/streams.py

```diff
@@ -802,51 +805,77 @@ def connect(self, src):
+
+    # Make the fifo to be used for commands
+    os.mkfifo(f'{self.src_config_folder}/cmd')
 
-    meta_args = [f'{utils.get_folder("streams")}/dlna_metadata.bash', f'{src_config_folder}']
+    # startup the metadata process and the DLNA process
+    meta_args = [f'{utils.get_folder("streams")}/dlna_meta.py',
+                 f'{self.name}',
+                 f'{self.src_config_folder}/cmd',
+                 f'{self.src_config_folder}/meta.json',
+                 self.src_web_folder, '-d']
     dlna_args = ['gmediarender', '--gstout-audiosink', 'alsasink',
                 '--gstout-audiodevice', utils.real_output_device(src), '--gstout-initial-volume-db',
                 '0.0', '-p', f'{portnum}', '-u', f'{self.uuid}',
-                '-f', f'{self.name}', '--logfile',
-                f'{src_config_folder}/metafifo']
-    self.proc = subprocess.Popen(args=meta_args, preexec_fn=os.setpgrp, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    self.proc2 = subprocess.Popen(args=dlna_args, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+                '-f', f'{self.name}']
+    self.metadata_proc = subprocess.Popen(args=meta_args)
+    self.dlna_proc = subprocess.Popen(args=dlna_args, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    self.fifo = os.open(f'{self.src_config_folder}/cmd', os.O_WRONLY, os.O_NONBLOCK) # open the fifo for writing but don't block in case something goes wrong
+
     self._connect(src)
 
   def disconnect(self):
     if self._is_running():
-      os.killpg(os.getpgid(self.proc.pid), signal.SIGKILL)
-      self.proc2.kill()
+      self.metadata_proc.kill()
+      self.dlna_proc.kill()
+      self.fifo.close()
     self._disconnect()
-    self.proc = None
-    self.proc2 = None
+    self.metadata_proc = None
+    self.dlna_proc = None
 
   def info(self) -> models.SourceInfo:
-    src_config_folder = f'{utils.get_folder("config")}/srcs/{self.src}'
-    loc = f'{src_config_folder}/currentSong'
     source = models.SourceInfo(name=self.full_name(), state=self.state, img_url='static/imgs/dlna.png')
     try:
-      with open(loc, 'r', encoding='utf-8') as file:
-        for line in file.readlines():
-          line = line.strip()
-          if line:
-            d = ast.literal_eval(line)
-        if 'state' in d:
-          source.state = d['state']
-        if 'album' in d:
-          source.album = d['album']
-        if 'artist' in d:
-          source.artist = d['artist']
-        if 'title' in d:
-          source.track = d['title']
-        return source
-    except Exception:
+      data = json.load(open(f'{self.src_config_folder}/meta.json'))
+      print(data)
+      source.state = data['state']
+      if source.state != 'stopped': # if the state is stopped, just use default values
+        source.artist = data['artist']
+        source.track = data['title']
+        source.album = data['album']
```

Comment:
We may want to add some defaults to these like:

```suggestion
        # handle unpopulated fields
        source.artist = data.get('artist')
        source.track = data.get('title')
        source.album = data.get('album')
````

Reply:
Actually based on how the metadata script works, this is probably not necessary.